pkg/proc/native/proc_freebsd.go

```diff
@@ -348,10 +373,34 @@ func (dbp *nativeProcess) resume() error {
 			thread.CurrentBreakpoint.Clear()
 		}
 	}
+	if len(dbp.os.trapThreads) > 0 {
+		// On these threads we have already received a SIGTRAP while stepping on a different thread,
+		// do not resume the process, instead set a breakpoint on them.
+		tt := dbp.os.trapThreads
+		dbp.os.trapThreads = dbp.os.trapThreads[:0]
+		var err error
+		dbp.os.selectedThread, err = dbp.stop2(tt...)
+		return err
+	}
 	// all threads are resumed
 	var err error
-	dbp.execPtraceFunc(func() { err = ptraceCont(dbp.pid, 0) })
-	return err
+	dbp.execPtraceFunc(func() {
+		for _, th := range dbp.threads {
+			err = ptraceResume(th.ID)
+			if err != nil {
+				return
+			}
+		}
+	})
+	if err != nil {
+		return err
+	}
+	sig := int(dbp.os.delayedSignal)
+	dbp.os.delayedSignal = 0
+	for _, thread := range dbp.threads {
+		thread.Status = nil
+	}
+	return dbp.ptraceCont(sig)
```

Comment:
Perhaps we should also batch this into the `execPtraceFunc` above to minimize overhead.